### PR TITLE
Add a refresh button to the editor file browser

### PR DESCRIPTION
**Pull request by Robyt3**
screenshot_2023-01-13_22-51-07

## Checklist

- [X] Tested the change ingame
- [X] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by heinrich5991** [comment]
This seems to keep the selection at the same index. Is that the intended behavior?

I could imagine staying at the same file being more useful. We could say it's too annoying to implement, even if we'd consider it better. What do you think?

**Comment by Robyt3** [comment]
The current selection is already kept based on the filename when refreshing by calling `RefreshFilteredFileList`, so if new items appear before the selection it gets moved.

Previously the selection was always reset to the first item after calling this function, so I added the `KeepSelection` parameter, so this is not done when refeshing.

File: src/game/editor/editor.cpp [hunk 1]
```diff
@@ -4629,11 +4636,14 @@ void CEditor::FilelistPopulate(int StorageType)
...
-	else
-		m_aFilesSelectedName[0] = '\0';
+	if(!KeepSelection)
+	{
+		m_FilesSelectedIndex = m_vpFilteredFileList.empty() ? -1 : 0;
+		if(m_FilesSelectedIndex >= 0)
+			str_copy(m_aFilesSelectedName, m_vpFilteredFileList[m_FilesSelectedIndex]->m_aName);
+		else
+			m_aFilesSelectedName[0] = '\0';
+	}
```
Comment (heinrich5991) [inline src/game/editor/editor.cpp:4646]: It seems to me that `m_FilesSelectedIndex` isn't updated here if `KeepSelection` is false. In this case, `m_FilesSelectedIndex` and `m_aFilesSelectedName` might disagree. That sounds wrong. Am I making a mistake? I don't understand all of the code.

Comment (Robyt3) [reply src/game/editor/editor.cpp:4646]: `RefreshFilteredFileList`, which is called before, always updated `m_FilesSelectedIndex` according to the current `m_aFilesSelectedName`, so the selection is synchronized.

**Comment by Robyt3** [comment]
This is still ready to be merged. The selection is properly kept both when refreshing adds new items and when refreshing removes existing items.

**Comment by heinrich5991** [comment]
Sorry for not getting back.

Looks good to me now.

bors r+

**Pull request merged**